lib.py

```diff
@@ -483,6 +518,84 @@ def convert_type(wdl_type: Any) -> Optional[WDLBase]:
             return None
         converted_inner_type = convert_type(inner_type)
         # if inner type conversion failed, then type is invalid, so return None
-        return outer_py_typ(converted_inner_type) if converted_inner_type is not None else None
+        return WDLArray(converted_inner_type, optional=optional, nonempty=nonempty) if converted_inner_type is not None else None
     # primitives remaining
-    return wdl_type_to_miniwdl_class(wdl_type)()
+    return wdl_type_to_miniwdl_class(wdl_type)(optional=optional)
+
+
+def test_gpu_available():
+    if bool(os.getenv("WDL_CONFORMANCE_TESTS__GPU")):
+        # override
+        return True
+    try:
+        p = subprocess.run("nvidia-smi".split(" "))
+    except (FileNotFoundError, subprocess.SubprocessError):
+        pass
+    else:
+        # we have an nvidia gpu
+        if p.returncode == 0:
+            return True
```

Comment:
A FileNotFoundError gets raised, which is in the try except block below.

Reply:
I don't think that can be true. I'm talking about the case where `nvidia-smi` exists and can be launched by `subprocess`, but then `nnvidia-smi` looks and finds that no physical Nvidia GPU is installed in the computer. Does it exit with a nonzero status (and cause a `SubprocessError` to be raised on the Python side), or does it exit with a successful zero status after printing a list of zero Nvidia GPUs?

It looks like the code should work as is, because in my testing `nvidia-smi` appears to exit with code 6 when it can talk to the driver but has no GPUs available:
```
[anovak@mustard lr-giraffe]$ srun --partition=gpu -c2 --mem 1G nvidia-smi
No devices were found
srun: error: phoenix-00: task 0: Exited with exit code 6
[anovak@mustard lr-giraffe]$ srun --partition=gpu -c2 --mem 1G -G 1 nvidia-smi
Tue Aug 20 15:10:25 2024
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 530.30.02              Driver Version: 530.30.02    CUDA Version: 12.1     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                  Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf            Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA A100-SXM4-80GB           On | 00000000:8A:00.0 Off |                    0 |
| N/A   29C    P0               64W / 400W|      0MiB / 81920MiB |      0%      Default |
|                                         |                      |             Disabled |
+-----------------------------------------+----------------------+----------------------+

+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|  No running processes found                                                           |
+---------------------------------------------------------------------------------------+
```

Reply:
I see, sounds like it should work in that case then